gno.land/pkg/integration/testing_integration.go

```diff
@@ -0,0 +1,257 @@
+						// availability. For now, if this sleep duration is too short, the
+						// subsequent command might fail with an [internal error].
+						time.Sleep(time.Second)
+					}
+				case "stop":
+					n, ok := nodes[sid]
+					if !ok {
+						err = fmt.Errorf("node %q not started cannot be stop", sid)
+						break
+					}
+
+					if err = n.Stop(); err != nil {
+						delete(nodes, sid)
+
+						// Unset gnoland environements.
+						ts.Setenv("RPC_ADDR", "")
+						ts.Setenv("GNODATA", "")
+					}
+				}
+
+				tsValidateError(ts, "gnoland "+cmd, neg, err)
+			},
+			"gnokey": func(ts *testscript.TestScript, neg bool, args []string) {
+				muNodes.Lock()
+				defer muNodes.Unlock()
+
+				sid := getSessionID(ts)
+
+				// Setup IO command.
+				io := commands.NewTestIO()
+				io.SetOut(commands.WriteNopCloser(ts.Stdout()))
+				io.SetErr(commands.WriteNopCloser(ts.Stderr()))
+				cmd := client.NewRootCmd(io)
+
+				io.SetIn(strings.NewReader("\n")) // Inject empty password to stdin.
+				defaultArgs := []string{
+					"-home", gnoHomeDir,
+					"-insecure-password-stdin=true", // There no use to not have this param by default.
+				}
+
+				if n, ok := nodes[sid]; ok {
+					if raddr := n.Config().RPC.ListenAddress; raddr != "" {
+						defaultArgs = append(defaultArgs, "-remote", raddr)
+					}
+
+					n.nGnoKeyExec++
+					headerlog := fmt.Sprintf("%.02d!EXEC_GNOKEY", n.nGnoKeyExec)
+					// Log the command inside gnoland logger, so we can better scope errors.
+					n.logger.Info(headerlog, strings.Join(args, " "))
+					defer n.logger.Info(headerlog, "END")
+				}
+
+				// Inject default argument, if duplicate
+				// arguments, it should be override by the ones
+				// user provided.
+				args = append(defaultArgs, args...)
+
+				err := cmd.ParseAndRun(context.Background(), args)
+
+				tsValidateError(ts, "gnokey", neg, err)
+			},
+		},
+	}
+}
+
+func getSessionID(ts *testscript.TestScript) string {
+	works := ts.Getenv("WORK")
+	sum := crc32.ChecksumIEEE([]byte(works))
+	return strconv.FormatUint(uint64(sum), 16)
+}
+
+func getTestingLogger(ts *testscript.TestScript, logname string) log.Logger {
+	var path string
+	if logdir := os.Getenv("LOG_DIR"); logdir != "" {
+		if err := os.MkdirAll(logdir, 0o755); err != nil {
+			ts.Fatalf("unable to make log directory %q", logdir)
+		}
+
+		var err error
+		if path, err = filepath.Abs(filepath.Join(logdir, logname)); err != nil {
+			ts.Fatalf("uanble to get absolute path of logdir %q", logdir)
+		}
+	} else if workdir := ts.Getenv("WORK"); workdir != "" {
+		path = filepath.Join(workdir, logname)
+	} else {
+		return log.NewNopLogger()
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		ts.Fatalf("unable to create log file %q: %s", path, err.Error())
+	}
+
+	ts.Defer(func() {
+		if err := f.Close(); err != nil {
+			panic(fmt.Errorf("unable to close log file %q: %w", path, err))
+		}
+	})
+
+	logger := log.NewTMLogger(f)
+	switch level := os.Getenv("LOG_LEVEL"); strings.ToLower(level) {
```

Comment:
other flag which needs to be documented, we can't expect everyone using txtar to look at the code

Reply:
I'm on it !